src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double RESIZE_COUNT = 1.5;
+    private Object[] elementData;
+    private int defaultSize;
+    private int count;
+
+    public ArrayList() {
+        defaultSize = 10;
+        elementData = new Object[defaultSize];
+        count = 0;
+    }
+
+    private boolean rangeCheckForCount(int index) {
+        if (index >= count || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound of array.");
+        }
+        return true;
+    }
+
+    private void checkArrayLength(int count) {
+        if (count >= defaultSize) {
+            Object[] localeData = elementData;
+            elementData = new Object[defaultSize *= RESIZE_COUNT];
+            System.arraycopy(localeData, 0, elementData, 0, localeData.length);
```

Comment:
As i understood you use here default size to compare if there enough space, but you never update it. I mean, this if statement always will be true, when count reach default size in value. And it's mean that you always create new element data. My advice - as i mentioned early - make DEFAULT_SIZE as constant, and change logic of this method by taking in consideration actual size of your elementData.

Reply:
Sorry, everything ok, i haven't noticed it : defaultSize *= RESIZE_COUNT. But again, it's better to make default size constant and don't change it in code.